lib/ret/hub.ex

```diff
@@ -897,6 +897,57 @@ end
 
 defimpl Canada.Can, for: Ret.Account do
   alias Ret.{Hub, AppConfig}
+  alias Ret.Api.{Scopes, Credentials}
+
+  # Non-admin accounts can create their own account tokens
+  def can?(%Ret.Account{account_id: account_id}, :create_credentials, %{
+        subject_type: :account,
+        scopes: scopes,
+        account_id: account_id
+      }) do
+    # TODO: Is it correct to disallow the create_accounts scope for account type tokens?
+    # If so, why? Do we need to enforce this everywhere?
+    Scopes.create_accounts() not in scopes
+  end
+
+  # Admin accounts can create tokens on behalf of other users
+  # TODO: Should we even allow this?
+  def can?(%Ret.Account{is_admin: true}, :create_credentials, %{subject_type: :account, scopes: scopes}) do
+    Scopes.create_accounts() not in scopes
+  end
+
+  def can?(%Ret.Account{is_admin: is_admin}, :create_credentials, %{subject_type: :app}) do
+    is_admin
+  end
+
+  def can?(%Ret.Account{}, :create_credentials, _params) do
+    false
+  end
+
+  def can?(%Ret.Account{is_admin: is_admin}, :list_credentials, :app) do
+    is_admin
+  end
+
+  def can?(%Ret.Account{}, :list_credentials, :account) do
+    # TODO: Allow admins to disable this in config
```

Comment:
I don't know how/where to add config values. I'll have to learn

Reply:
Added in https://github.com/mozilla/reticulum/pull/436/commits/f52a5c2ddcb702b3d2aa1f1c0b4f401bc7de4009 and https://github.com/mozilla/hubs/pull/3551